ignite/cmd/network_chain_join.go

```diff
@@ -175,9 +175,25 @@ func askPublicAddress(ctx context.Context, session cliui.Session) (publicAddress
 
 	// even if GetIp fails we won't handle the error because we don't want to interrupt a join process.
 	// just in case if GetIp fails user should enter his address manually
-	ip, err := ipify.GetIp()
-	if err == nil {
-		options = append(options, cliquiz.DefaultAnswer(fmt.Sprintf("%s:26656", ip)))
+	peerAddress, _ := cmd.Flags().GetString(flagPeerAddress)
+	if peerAddress == "" {
+		if ip, err := ipify.GetIp(); err == nil {
+			peerAddress = fmt.Sprintf("%s:26656", ip)
+		}
+	}
+
+	// Use the current peer address when "--yes" flag is present
+	if getYes(cmd) {
+		if peerAddress == "" {
+			return "", errors.New("a peer address is required")
+		}
+
+		return peerAddress, nil
+	}
+
+	options := []cliquiz.Option{cliquiz.Required()}
+	if peerAddress != "" {
+		options = append(options, cliquiz.DefaultAnswer(peerAddress))
```

Comment:
Unlike specified in the PR body, I'm not sure that passing the `peer-address` flag avoids the prompt, unless `-y` is also passed. Maybe this is intended?

Reply:
Yes the prompt still actually exists but with the provided address as the default
```
ignite n chain join 27 --peer-address https://toto.com:26657
? Peer's address https://toto.com:26657
Source code fetched
Blockchain set up
```

Reply:
Yes, when the peer address is given as a flag value it is used as default, but as recommended prompt won't appear when the flag is setted as it makes not much sense to ask again in that case.